### ISSUE TITLE
[BUG] Debian OpenJDK breaks Java Version Parser

### ISSUE THREAD

**Issue by maximaximal** [bloodmc/GriefDefender#426]
Minecraft 1.17.1, git-Paper-327, GriefDefender v2.0.1, LuckPerms v5.3.47

When on Debian 11 and using the `openjdk-17-jre` package, then `java -version` reports the following:

```
openjdk version "17-ea" 2021-09-14
OpenJDK Runtime Environment (build 17-ea+19-Debian-1)
OpenJDK 64-Bit Server VM (build 17-ea+19-Debian-1, mixed mode, sharing)
```

Sadly, the current Java version parser breaks because of the `-ea` suffix with the following exception(s):

```
[14:27:44 WARN]: java.lang.NumberFormatException: For input string: "17-ea"
[14:27:44 WARN]: 	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
[14:27:44 WARN]: 	at java.base/java.lang.Integer.parseInt(Integer.java:668)
[14:27:44 WARN]: 	at java.base/java.lang.Integer.parseInt(Integer.java:786)
[14:27:44 WARN]: 	at com.griefdefender.GDBootstrap.getJavaVersion(GDBootstrap.java:251)
[14:27:44 WARN]: 	at com.griefdefender.GDBootstrap.onEnable(GDBootstrap.java:125)
[14:27:44 WARN]: 	at griefdefender-bukkit-2.0.1.jar//com.griefdefender.loader.BukkitLoaderPlugin.onEnable(BukkitLoaderPlugin.java:48)
[14:27:44 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264)
[14:27:44 WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
[14:27:44 WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:502)
[14:27:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:544)
[14:27:44 WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:458)
[14:27:44 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:288)
[14:27:44 WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1212)
[14:27:44 WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319)
[14:27:44 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
[14:27:44 ERROR]: Error occurred while enabling GriefDefender v2.0.1 (Is it up to date?)
java.lang.NoClassDefFoundError: com/griefdefender/lib/configurate/util/NamingScheme
	at com.griefdefender.GDBootstrap.onEnable(GDBootstrap.java:151) ~[?:?]
	at com.griefdefender.loader.BukkitLoaderPlugin.onEnable(BukkitLoaderPlugin.java:48) ~[griefdefender-bukkit-2.0.1.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-327]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-327]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:502) ~[patched_1.17.1.jar:git-Paper-327]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:544) ~[patched_1.17.1.jar:git-Paper-327]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:458) ~[patched_1.17.1.jar:git-Paper-327]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:288) ~[patched_1.17.1.jar:git-Paper-327]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1212) ~[patched_1.17.1.jar:git-Paper-327]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-327]
	at java.lang.Thread.run(Thread.java:831) ~[?:?]
Caused by: java.lang.ClassNotFoundException: com.griefdefender.lib.configurate.util.NamingScheme
	at java.net.URLClassLoader.findClass(URLClassLoader.java:432) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:586) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?]
	... 11 more
```

The problematic code in question seems to be this one:

https://github.com/bloodmc/GriefDefender/blob/e281231d874d78def944e748f6b49057a516b3db/bukkit/src/main/java/com/griefdefender/GDBootstrap.java#L239

Thank you for looking into this and best regards!

**Comment by bloodmc** [comment]
Thanks, I'll have this fixed next release. For now, don't use any early access builds and it should work fine.

**Comment by bloodmc** [comment]
Fixed in GD 2.0.3

**Comment by maximaximal** [comment]
Very nice, thank you for the support!
